### ISSUE TITLE
Problems about shape input and output for the models

### ISSUE THREAD

**Issue by Strawbili** [Miche11eU/PET-Image-Denoising-Using-3D-Diffusion-Model#2]
Dear Maintainers,

Your work is really impressive, and it turned out to be a unified model for many clinial PET images. But I do have a question about the input and the output shape of the models.

In your work, it seems like the input shape of the image is (192, 288,  520)
```
with bf.BlobFile(base_samples, "rb") as f:
        obj = np.load(f)
        low_pet = obj["arr_0"][0] 
        image_arr = np.zeros((6,192,288,96))
        index = 0
        for i in (0, 86, 172, 258, 344, 424):
            image_arr[index,:,:,:] = low_pet[:, :, i:i+96]
            index += 1
```
However, it seems like the output shape is (192,288,576)
```
arr_result = np.zeros((192,288,576))     
    index = 0
    for i in range(6):
        arr_result[:, :, i*96:(i+1)*96] = arr[index,:,:,:]
        index += 1
```

Is it a bug, or you do them in purpose? 

I have a PET image with 159 frames in the axial direction. So depending on the current situation, I should simply divide them by 2?

Best,
Yuhan

**Comment by Miche11eU** [comment]
Hi Yuhan, @Strawbili 

Thank you very much for pointing out this important question.

To denoise large 3D PET volumes with limited GPU memory, we split the input image along the axial dimension into 6 patches overlapped sequentially by 10 pixels,  each with a shape of [192, 288, 96]. These patches were stitched together after inference into a temporary array of size [192, 288, 576]. The output was not yet combined in the earlier script, which caused the confusion.

To ensure smooth transitions between adjacent patches, we applied a weighted average in the overlapping areas. This logic has been added to the function reverse_2_original_dimension() and integrated into the latest version of scripts/test.py.

For your PET image with 159 axial slices, you can divide it into 2 patches, each with a depth of 96, and let them overlap by 33 pixels.

Please refer to the updated README and modify the load_data_for_worker() and reverse_2_original_dimension() functions accordingly to match your input shape and desired overlap.

Feel free to reach out if you’d like further help with this setup.

Best,
Boxiao

**Comment by Miche11eU** [comment]
Hi Jimmy @Jimmy-Bots ,

Thank you for your interest and detailed feedback.

For metric calculation, I’ve added an evaluation script under evaluation/get_ssim_psnr.py, which includes our exact implementation for computing SSIM and PSNR. You can find usage instructions and detailed comments in the script. Please make sure that your preprocessing and postprocessing steps are consistent with those outlined in the code. Inconsistencies in any of these steps could affect the final metric values.

For model training, the optimizer was set to AdamW, the learning rate was set to 1e-4, and the MSE loss was used. Paired normal-dose and 1/20 low-dose PET images were used as the training data.

Your current 3D DDPM result appears worse than expected, could you share more details about your preprocessing pipeline? For example, did you convert your input data to SUV units before inference? How did you normalize the data?

I have now updated the inference script (scripts/test.py) to include logic for overlapping patches and weighted averaging. Please check the updated versions of load_data_for_worker and reverse_2_original_dimension functions and the README to ensure your inference setup is aligned.

Since the 3D DDPM model was trained on paired 1/20 low-dose and normal-dose PET, you can also verify performance specifically on 1/20 low-dose inputs to ensure your inference setup is appropriate.

Please let me know if you'd like further help with this setup.

Best,
Boxiao

**Comment by Strawbili** [comment]
Hi Boxiao,

Sorry for the late reply. The code turned out to be great!

We are doing something base on your work. Hope to share some exciting progress in the near future!

Best,
Yuhan
